src/app/content/practiceQuestions/utils.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { book } from '../../../test/mocks/archiveLoader';
+import { findArchiveTreeNodeById } from '../utils/archiveTreeUtils';
+import { stripIdVersion } from '../utils/idUtils';
+import { PracticeQuestionsSummary } from './types';
+import {
+  getNextPageWithPracticeQuestions,
+  getPracticeQuestionsLocationFilters,
+  pageHasPracticeQuestions,
+} from './utils';
+
+describe('pageHasPracticeQuestions', () => {
+  it('returns true', () => {
+    expect(pageHasPracticeQuestions('pageId', { countsPerSource: { pageId: 2, other: 0 } })).toEqual(true);
+  });
+
+  it('returns false', () => {
+    expect(pageHasPracticeQuestions('asdasd', { countsPerSource: { pageId: 2, other: 0 } })).toEqual(false);
+  });
+});
+
+describe('getPracticeQuestionsLocationFilters', () => {
+  it('returns empty map if book or summary is not truthy', () => {
+    expect(getPracticeQuestionsLocationFilters(null, book)).toEqual(new Map());
+    expect(getPracticeQuestionsLocationFilters({ countsPerSource: {} }, undefined)).toEqual(new Map());
+  });
+
+  it('returns location filters for practice questions', () => {
+    const page1 = (book as any).tree.contents[1].contents[0].contents[0]; // testbook1-testpage2-uuid
+    const linkedPage1 = findArchiveTreeNodeById(book.tree, 'testbook1-testpage2-uuid');
```

Comment:
You could just use `linkedPageX` everywhere, right? Just need to return with assertion if its undefined / assertDefined